### PR TITLE
Support archiving subrecipients

### PR DESCRIPTION
**Pull request by ajhyndman**
## Description

Implements support for "archiving" subrecipients.

An "archived" subrecipient will be excluded from treasury export generation.  For all other purposes, an "archived" subrecipient still behaves identically.  It is still displayed in the UI, and can still be edited, and will still be considered for uniqueness constraints for future uploads.

## Screenshots / Demo Video

image

image

## Testing

- `yarn db:migrate`
- `yarn start`
- If not already present, upload a workbook that includes at least one subrecipient
- Trigger a treasury export, _verify that subrecipient is present_
- Navigate to "Subrecipients" tab
- Click "archive", _verify that subrecipient is updated in this view_
- Trigger a treasury export, _verify that subrecipient is no longer present_

### Automated and Unit Tests
- [ ] Added Unit tests

### Manual tests for Reviewer
- [x] Added steps to test feature/functionality manually

## Checklist
- [x] Provided ticket and description
- [x] Provided screenshots/demo
- [x] Provided testing information
- [x] Provided adequate test coverage for all new code
- [x] Added PR reviewers

### CODE REVIEW

File: packages/client/src/arpa_reporter/views/Subrecipients.vue [hunk 1]
```diff
@@ -23,31 +29,35 @@
...
-        </span>
-
-        <span v-else>
-          {{props.formattedRow[props.column.field]}}
-        </span>
+          </ul>
+
+          <span class="d-flex flex-column" v-else-if="props.column.field === 'edit'" >
+            <router-link v-if="!props.row.is_archived" tag="button" class="btn btn-sm btn-secondary mb-2" :to="`/subrecipients/${props.row.id}`">Edit</router-link>
+            <button v-if="props.row.is_archived" class="btn btn-sm btn-primary" @click="archiveSubrecipient(props.row.id)">Restore</button>
```
Comment (tzinckgraf) [inline packages/client/src/arpa_reporter/views/Subrecipients.vue:53]: Having one function can be fine, but can we update the naming? Maybe something like `toggleArchiveSubrecipient`. It was confusing to read at first when the `archive` function can both archive and restore. Thoughts?

File: packages/server/src/arpa_reporter/db/arpa-subrecipients.js [hunk 2]
```diff
@@ -14,6 +14,21 @@ function baseQuery(trns) {
         .leftJoin('users AS users2', 'arpa_subrecipients.updated_by', 'users2.id');
 }
 
+async function archiveRecipient(id, { updatedByUser }, trns = knex) {
+    const query = trns('arpa_subrecipients')
+        .where('id', id)
+        .returning('*');
+
+    if (updatedByUser) {
```
Comment (tzinckgraf) [inline packages/server/src/arpa_reporter/db/arpa-subrecipients.js:22]: Should this be required? At the very least, we might want to always update the `updated_at` field.

File: packages/client/src/arpa_reporter/views/Subrecipients.vue (hunk 1 above)
Comment (ajhyndman) [reply packages/client/src/arpa_reporter/views/Subrecipients.vue:53]: Yeah, good note.  I definitely cut a corner there.  I do like the naming suggestion.

File: packages/server/src/arpa_reporter/db/arpa-subrecipients.js (hunk 2 above)
Comment (ajhyndman) [reply packages/server/src/arpa_reporter/db/arpa-subrecipients.js:27]: I don't have a strong opinion here.  I just copied this pattern from the `updateRecipient` method in the same file below.  I figure they should at least be consistent with each other.

I did mark session as required in the API endpoint though, so it should always be set in practice.

**Review by tzinckgraf — APPROVED** [review]

**Comment by as1729** [comment]
@ajhyndman I think there might be an issue with the datetime stamp on the migrations since there was another migration merged prior to this.

File: packages/server/src/arpa_reporter/db/arpa-subrecipients.js [hunk 3]
```diff
@@ -70,10 +93,11 @@ async function listRecipients(trns = knex) {
 }
 
 async function listRecipientsForReportingPeriod(periodId, trns = knex) {
-    return baseQuery(trns).where('reporting_period_id', periodId);
+    return baseQuery(trns).where('reporting_period_id', periodId).where('is_archived', false);
```
[the remark this answers is not in the document]
Comment (ajhyndman) [reply packages/server/src/arpa_reporter/db/arpa-subrecipients.js:96]: Thanks for the catch!

File: packages/client/src/arpa_reporter/views/Subrecipients.vue [hunk 4]
```diff
@@ -71,7 +81,9 @@ export default {
   },
   computed: {
     rows() {
-      return this.recipients.map((r) => ({ ...r, json: JSON.parse(r.record) }));
+      const rows = this.recipients.map((r) => ({ ...r, json: JSON.parse(r.record) }));
+      console.log(rows);
```
Comment (tzinckgraf) [inline packages/client/src/arpa_reporter/views/Subrecipients.vue:85]: do we want to leave the `console.log` in here?

**Review by tzinckgraf — APPROVED** [review]

**Pull request merged**